### DISCUSSION TITLE
Build on top of Rails 7

### DISCUSSION THREAD

**Discussion by bradgessler** [bridgetownrb/bridgetown#470] [Ideas]
Greetings! I've been hacking on a Ruby static site generator over the years that's a lot like Bridgetown called [Sitepress](https://sitepress.cc/).

Half way through I had a revelation that I was rewriting a lot of stuff that already shipped with Rails, so I scrapped a lot of my code and replaced it with a subset of Rails. I'm curious if Bridgetown contributors have run into that problem yet?

One benefit of this approach is that Sitepress can run stand-alone, just like Bridgetown, Middleman, & Jekyll or it can be embedded into a Rails application and serve up pages dynamically.

I'm hoping when Rails 7 gets released, I can ride off the improvements in choices they provide for asset management (esbuild, webpacker, no-build tool, etc).

Anyway, just wanted to say hi and see how others are thinking about this. Bridgetown looks pretty awesome and it's always great to see folks building quality software.

**Comment by jaredcwhite** [comment]
Thanks for the question @bradgessler. We're deeply indebted to Rails for what it's brought to the Ruby community, and we make use of Active Support and Zeitwek in Bridgetown, as well as mirror Rails' handling of ERB syntax.

However, Bridgetown is definitely its own project with its own goals and perspective. I personally don't like the way Rails 7 deals with frontend bundling on a number of fronts and believe what we'll be shipping soon in v1.0 is far superior. I'm also very bullish on using Roda for dynamic routes in smaller projects which don't require all the conceptual overhead of Rails. So there's some divergence in thinking here, and I like that. A rising tide floats all boats, as they say. 😃

**Reply by bradgessler** [reply]
Yeah I'm in the middle of getting the asset pipeline to work with Rails 7 and TBH it's not super great. The choice is nice, but it's now way more difficult for third party gems to plug their assets cleanly into whatever Rails chooses. I'll keep an eye on this project and see what you come up with; I'm sure I'll be tempted to roll it into Sitepress.
